pppp.c

```diff
@@ -497,16 +534,44 @@ int startup()
 	printf("HTTP connection from %s:%d\n",
 		inet_ntoa(iclientip->sin_addr), ntohs(iclientip->sin_port));
 	while (1) {
-		len = read(client, inbuf, sizeof(inbuf));
-		if (len < 7) {
-			close(client);
-			goto again;
-		}
-		if (strncasecmp(inbuf, "GET ", 4)) {
-			close(client);
-			goto again;
-		}
-		if (!strncmp(inbuf+4, "/v.mjpg ", 8)) {
+        char inbuf[1024];
+        int len = read(client, inbuf, sizeof(inbuf));
+        if (len <= 7) {
+            close(client);
+            goto again;
+        }
+		// if (strncasecmp(inbuf, "GET ", 4)) {
+		// 	close(client);
+		// 	goto again;
+		// }
+        if (!strncmp(inbuf+4, "/exec ", 7)) {
+            char *json_end = strstr(inbuf+7, "\r\n");
+            if (json_end) {
+                *json_end = '\0'; // Null terminate the JSON string
+                sendExecCmd(inbuf+7);
```

Comment:
you know the start and end of the command string here, so you can easily compute its length. You should do so and pass it as a parameter to sendExecCmd, instead of using strlen() there. Using strlen() is nearly always a sign of poorly written code.